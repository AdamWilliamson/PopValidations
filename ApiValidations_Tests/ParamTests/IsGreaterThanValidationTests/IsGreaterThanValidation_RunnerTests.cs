using ApprovalTests;
using FluentAssertions;
using ApiValidations;
using ApiValidations_Tests.TestHelpers;

namespace ApiValidations_Tests.ValidationsTests.IsGreaterThanValidationTests;

public class IsGreaterThanApi
{
    public void SetValues(int param1, string param2, decimal param3, double param4, short param5, long param6) { }
}

public class IsGreaterThan_TestingValidator : ApiValidator<IsGreaterThanApi>
{
    public IsGreaterThan_TestingValidator()
    {
        DescribeFunc(x => x.SetValues(
            Param.Is<int>().IsGreaterThan(int.MaxValue),
            Param.Is<string>().IsGreaterThan(new string(char.MaxValue, 100)),
            Param.Is<decimal>().IsGreaterThan(decimal.MaxValue),
            Param.Is<double>().IsGreaterThan(double.MaxValue),
            Param.Is<short>().IsGreaterThan(short.MaxValue),
            Param.Is<long>().IsGreaterThan(long.MaxValue)
        ));
    }
}

public class IsGreaterThanValidation_RunnerTests
{
    [Fact]
    public void WhenDescribing_ItReturnsTheValidation()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new IsGreaterThan_TestingValidator());

        // Act
        var description = runner.Describe();

        // Assert
        description.Results.Should().HaveCount(6);
        description.Results.Should().HaveCount(ValidatableHelper.GetValidatableCount<IsGreaterThanApi>(ValidatableType.NoExceptions));
        Approvals.VerifyJson(JsonConverter.ToJson(description));
    }
}
