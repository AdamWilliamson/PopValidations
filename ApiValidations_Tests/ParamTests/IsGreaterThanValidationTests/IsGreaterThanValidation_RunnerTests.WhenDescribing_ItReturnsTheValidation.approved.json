{
    "results": [
        {
            "property": "SetValues(Int32,String,Decimal,Double,Int16,Int64):Void::(param1,0,Int32)",
            "outcomes": [
                {
                    "validator": "IsGreaterThanValidation",
                    "message": "Must be greater than '2147483647'.",
                    "values": [
                        {
                            "key": "value",
                            "value": "2147483647"
                        }
                    ]
                }
            ],
            "validationGroups": []
        },
        {
            "property": "SetValues(Int32,String,Decimal,Double,Int16,Int64):Void::(param2,1,String)",
            "outcomes": [
                {
                    "validator": "IsGreaterThanValidation",
                    "message": "Must be greater than '￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿'.",
                    "values": [
                        {
                            "key": "value",
                            "value": "￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿"
                        }
                    ]
                }
            ],
            "validationGroups": []
        },
        {
            "property": "SetValues(Int32,String,Decimal,Double,Int16,Int64):Void::(param3,2,Decimal)",
            "outcomes": [
                {
                    "validator": "IsGreaterThanValidation",
                    "message": "Must be greater than '79228162514264337593543950335'.",
                    "values": [
                        {
                            "key": "value",
                            "value": "79228162514264337593543950335"
                        }
                    ]
                }
            ],
            "validationGroups": []
        },
        {
            "property": "SetValues(Int32,String,Decimal,Double,Int16,Int64):Void::(param4,3,Double)",
            "outcomes": [
                {
                    "validator": "IsGreaterThanValidation",
                    "message": "Must be greater than '1.7976931348623157E+308'.",
                    "values": [
                        {
                            "key": "value",
                            "value": "1.7976931348623157E+308"
                        }
                    ]
                }
            ],
            "validationGroups": []
        },
        {
            "property": "SetValues(Int32,String,Decimal,Double,Int16,Int64):Void::(param5,4,Int16)",
            "outcomes": [
                {
                    "validator": "IsGreaterThanValidation",
                    "message": "Must be greater than '32767'.",
                    "values": [
                        {
                            "key": "value",
                            "value": "32767"
                        }
                    ]
                }
            ],
            "validationGroups": []
        },
        {
            "property": "SetValues(Int32,String,Decimal,Double,Int16,Int64):Void::(param6,5,Int64)",
            "outcomes": [
                {
                    "validator": "IsGreaterThanValidation",
                    "message": "Must be greater than '9223372036854775807'.",
                    "values": [
                        {
                            "key": "value",
                            "value": "9223372036854775807"
                        }
                    ]
                }
            ],
            "validationGroups": []
        }
    ]
}